### PR TITLE
Initial documentation for the TestAccessor Pattern

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by Youssef1313** [comment]
@sharwell What is the status for this?

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
Personally I'd avoid using spaces for new files.

File: docs/TestAccessor Pattern.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  {
+    private readonly SomeProductionType _instance;
+
+    internal TestAccessor(SomeProductionType instance)
+    {
+      _instance = instance;
+    }
+
+    internal ref int PrivateStateData => ref _instance._privateStateData;
+  }
```
Comment (Youssef1313) [inline docs/TestAccessor Pattern.md:34]:
```suggestion
    private int _privateStateData;

    internal readonly struct TestAccessor
    {
        private readonly SomeProductionType _instance;

        internal TestAccessor(SomeProductionType instance)
        {
            _instance = instance;
        }

        internal ref int PrivateStateData => ref _instance._privateStateData;
    }
```

File: docs/TestAccessor Pattern.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+```
+
+## The `GetTestAccessor()` Method
+
+The `GetTestAccessor()` method is always defined as follows:
+
+```csharp
+internal TestAccessor GetTestAccessor()
+  => new TestAccessor(this);
```
Comment (Youssef1313) [inline docs/TestAccessor Pattern.md:44]:
```suggestion
    => new TestAccessor(this);
```

File: docs/TestAccessor Pattern.md (hunk 1 above)
Comment (sharwell) [reply docs/TestAccessor Pattern.md:34]: This is intentionally on a 2-space indent for readability in a text document.

File: docs/TestAccessor Pattern.md (hunk 2 above)
Comment (sharwell) [reply docs/TestAccessor Pattern.md:44]: ➡️ This is intentionally on a 2-space indent for readability in a text document.

**Comment by sharwell** [comment]
> Personally I'd avoid using spaces in new files' names.

I was just following the pattern set by other files in the folder.

**Pull request merged**